File: jobs/2026-07-03__03-23-57/encore_multi_service_api_ts__DoUQvhk/artifacts/user/myapp/order/order.ts
import { api } from "encore.dev/api";
import { user } from "~encore/clients";

interface OrderResponse {
  orderId: number;
  userId: number;
  userName: string;
}

export const getOrder = api(
  { expose: true, method: "GET", path: "/order/:id" },
  async ({ id }: { id: number }): Promise<OrderResponse> => {
    // Call the user service internally to fetch the userName, hardcoded to ID 1.
    const userRes = await user.getUser({ id: 1 });
    return {
      orderId: id,
      userId: userRes.id,
      userName: userRes.name,
    };
  }
);

File: jobs/2026-07-03__12-25-47/encore_multi_service_api_go__LrMLYjV/artifacts/user/myproject/order/order.go
package order

import (
"context"

"myapp/user"
)

type Order struct {
ID       string `json:"id"`
UserID   string `json:"user_id"`
UserName string `json:"user_name"`
}

//encore:api public path=/order/:id method=GET
func GetOrder(ctx context.Context, id string) (*Order, error) {
u, err := user.GetUser(ctx, id)
if err != nil {
return nil, err
}
return &Order{
ID:       id,
UserID:   u.ID,
UserName: u.Name,
}, nil
}
